File: pset1/cash/cash.c
/**
 * CS50x 2021
 * Problem Set 1: cash
 */

#include <stdio.h>
#include <math.h>
#include <cs50.h>


int main(void)
{
    int cents;
    int count_coins(int n);

    // Prompt user for non-negative change owed.
    do {
        cents = round(get_float("Change owed: ") * 100);
    } while (cents < 0);

    printf("%i\n", count_coins(cents));

    return 0;
}

int count_coins(int n)
{
    int base, counts = 0;
    int coin_types[4] = {25, 10, 5, 1};

    if (!n)
      return 0;

    for (int i = 0; i < 4; i++) {
        base = coin_types[i];

        while (n >= base) {
            n -= base;
            counts++;
        }
    }

    return counts;
}

File: week6/pset6/cash/cash.py
from cs50 import get_float


def get_cash():
    """Prompt user for cash."""
    while True:
        cash = get_float("Change owed: ")
        if cash > 0.0:
            return cash


def count_coins(cash):
    # Convert cash to pennies.
    cash = int(cash*100)

    coins = {
        25: 0,  # quarters
        10: 0,  # dimes
        5: 0,  # nickels
        1: 0,  # pennies
    }

    for coin in coins:
        while cash >= coin:
            coins[coin] += 1
            cash -= coin

    return sum(coins.values())


def main():
    cash = get_cash()
    total_coins = count_coins(cash)
    print(total_coins)


if __name__ == "__main__":
    main()
